--- a/deps/jido/lib/jido/bus/adapters/in_memory.ex
+++ b/deps/jido/lib/jido/bus/adapters/in_memory.ex
@@ -456,12 +456,15 @@ defmodule Jido.Bus.Adapters.InMemory do
                                   now,
                                   %Signal{} = signal
                            ) do
               %Signal{
                      id: correlation_id,
                      source: causation_id,
                      type: type,
-      data: data,
-      jido_metadata: jido_metadata
+      data: data
               } = signal
 
+    jido_metadata = Map.get(signal, :jido_metadata, %{})
+
               %RecordedSignal{
                      signal_id: Jido.Util.generate_id(),
                      signal_number: signal_number,
                      stream_id: stream_id,
                      stream_version: stream_version,
@@ -471,7 +474,7 @@ defmodule Jido.Bus.Adapters.InMemory do
                      correlation_id: correlation_id,
                      type: type,
                      data: data,
-      jido_metadata: jido_metadata,
+      jido_metadata: jido_metadata || %{},
                      created_at: now
               }
        end
